workspace-run.js

```diff
@@ -42,36 +42,37 @@ function consume(cmds) {
 run({ name: 'yarn', args: ['workspaces', '--silent', 'info'] })
 	.then(info => JSON.parse(info))
 	.then(workspaceInfo => {
-		const orderedWorkspaceInfo = Object.entries(workspaceInfo).reduce(
-			(accu, [packageName, packageInfo]) => {
-				const { commands, packages } = accu;
-				const { location, workspaceDependencies } = packageInfo;
+		function add(acc, pkg) {
+			if (acc.indexOf(pkg) !== -1) {
+				return acc;
+			}
+			const wd = workspaceInfo[pkg].workspaceDependencies;
+
+			wd.reduce((kcc, dep) => {
+				add(kcc, dep);
```

Comment:
reccurisve add dependencies will make the script fails with a maximum recursion error so enforce we can't have one